driver.k

```diff
@@ -204,6 +208,12 @@ rule <k> #check "post" : {"mem" : #exists(S:String, V:Value) Ms:MemStatusList, R
      <env> ... String2Id(S) |-> #storedVar( I:Int , _ , _)... </env>
      <localMem> ... I |-> V ... </localMem>
 
+// This shouldn't be needed I think... But it gets stuck if removed. 
+rule <k> #check "post" : {"mem" : #exists(S:String, V:Value) Ms:MemStatusList, .MemStatusList}
+                 => #check "post" : {"mem" : Ms} ... </k>
+     <env> ... String2Id(S) |-> #storedVar( I:Int , _ , _)... </env>
+     <localMem> ... I |-> V ... </localMem>
+
```

Comment:
Yes, this looks nice, I changed so that it looks more like this. 
Btw, I also replaced that `#failure` with `#success`, as I think that was the intended semantics.

Reply:
Well, the idea is that if the test fails then it leaves the `#failure ...` in the K cell as the second element, but if the `#failure ...` makes it to the front of the K cell then it just goes away. So it's just a way to mark what the test-name that failed actually was (this is in the context of EVM where there are many tests per one file and it's useful to know exactly which one failed).

Either make sense I suppose, which probably means that neither make sense. But I guess I would like to keep it similar to KEVM if possible (which uses `#failure ...`).

Reply:
Ah I see. I wasn't thinking about KEVM when I made that comment. 
I just got confused by

```
rule <k> #failure TESTID => . ... </k>
     <mode> _ => SUCCESS </mode>
``` 

(from your comment above), so I thought that in this particular scenario it would make more sense to use `#success`. 

But in the context that you describe it makes sense (e.g. for keeping track of which test failed if we have many etc.). Actually I think it's a good idea to keep it "in sync" with KEVM (it's a point we discussed many times before). 

I am going to revert to `#failure` before merging.